.github/workflows/snyk.yml

```diff
@@ -94,6 +102,9 @@ jobs:
           export VERSION_TAG=$([[ "${VERSION}" == *-SNAPSHOT ]] && echo 'development' || echo "${VERSION}")
           export LIFECYCLE=$([[ "${VERSION}" == *-SNAPSHOT ]] && echo 'development' || echo 'production')
 
+          # use inputs.dockerImage if present; otherwise if building, use the output of the build step; otherwise infer from version and default repository camunda/zeebe
+          export DOCKER_IMAGE=$([[ ! -z '${{ inputs.dockerImage }}' ]] && echo '${{ inputs.dockerImage }}' || ([[ '${{ inputs.build }}' == 'true' ]] && echo '${{ steps.build-docker.outputs.image }}' || echo "camunda/zeebe:${VERSION}"))
```

Comment:
Tested locally via bash with the following combinations:

No `inputs.dockerImage`, but `inputs.build` is true.

```bash
export DOCKER_IMAGE=$([[ ! -z '' ]] && echo '' || ([[ 'true' == 'true' ]] && echo ' steps.build-docker.outputs.image' || echo "camunda/zeebe:${VERSION}")); echo $DOCKER_IMAGE
```

Prints 'steps.build-docker.outputs.image'


No `inputs.dockerImage`, and `inputs.build` is false.

```bash
export VERSION="8.1.0"
export DOCKER_IMAGE=$([[ ! -z '' ]] && echo '' || ([[ 'false' == 'true' ]] && echo ' steps.build-docker.outputs.image' || echo "camunda/zeebe:${VERSION}")); echo $DOCKER_IMAGE
```

Prints 'camunda/zeebe:8.1.0'

A given `inputs.dockerImage`, and `inputs.build` is true.

```bash
export VERSION="8.1.0"
export DOCKER_IMAGE=$([[ ! -z 'myImage' ]] && echo 'myImage' || ([[ 'true' == 'true' ]] && echo ' steps.build-docker.outputs.image' || echo "camunda/zeebe:${VERSION}")); echo $DOCKER_IMAGE
```

Prints 'myImage'


A given `inputs.dockerImage`, but `inputs.build` is false.

```bash
export VERSION="8.1.0"
export DOCKER_IMAGE=$([[ ! -z 'myImage' ]] && echo 'myImage' || ([[ 'false' == 'true' ]] && echo ' steps.build-docker.outputs.image' || echo "camunda/zeebe:${VERSION}")); echo $DOCKER_IMAGE
```

Prints 'myImage'

Reply:
I'm convinced, thanks for making the review easier :D